src/archetype/ArchetypeModule.ts

```diff
@@ -89,14 +93,25 @@ export namespace ArchetypeModule {
         }
     }
 
-    export async function updateArchetypeCatalog(): Promise<void> {
-        const xml: string = await Utils.downloadFile(REMOTE_ARCHETYPE_CATALOG_URL, true);
-        const archetypes: Archetype[] = await listArchetypeFromXml(xml);
+    export async function updateArchetypeCatalogs(): Promise<void> {
         const targetFilePath: string = path.join(getPathToExtensionRoot(), "resources", "archetypes.json");
+        const catalogs: ArchetypeCatalog[] | undefined = Settings.Archetype.catalogs();
+        let archetypes: Archetype[] = [];
+        archetypes = archetypes.concat(await parseArchetypeCatalog(REMOTE_ARCHETYPE_CATALOG_URL));
+        if (catalogs && !(_.isEmpty(catalogs))) {
+            for (const catalog of catalogs) {
+                archetypes = archetypes.concat(await parseArchetypeCatalog(catalog.url));
+            }
+        }
         await fse.ensureFile(targetFilePath);
         await fse.writeJSON(targetFilePath, archetypes);
     }
 
+    async function parseArchetypeCatalog(url: string): Promise<Archetype[]> {
+        const xml: string = await Utils.downloadFile(url, true);
```

Comment:
If the downloading fails, e.g. due to a wrong url or networking issue, an exception will be thrown. Thus even though the latest list of Maven Central archetypes is successfully retrieved, it's not updated into the local JSON file. Is that the behavior you want? 
* You thrown the expection to inform user about the error?
or 
* You swallow the exception and return an empty list for the malformed catalog silently?